maintenance/provision-lxc-node.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+set -ex
+source ./maintenance/helpers-lxc-node.sh
+
+container=${NODE_NAME:-"juju-client-box"}
+
+lxc launch ubuntu:16.04 $container
+
+while ! lxc info $container | grep -qE 'eth0:\sinet\s'; do
+    sleep 3
+done
+
+RUN add-apt-repository -y ppa:juju/stable
+RUN apt update -yq
+RUN apt install -y juju unzip python3-pip python-pip charm-tools squashfuse snapd
+RUN pip2 install 'git+https://github.com/juju-solutions/bundletester' \
+                 'git+https://github.com/juju-solutions/cloud-weather-report' \
+                 'git+https://github.com/juju/juju-crashdump'
+RUN pip3 install 'git+https://github.com/juju/python-libjuju' \
+                 'git+https://github.com/juju-solutions/matrix' \
+                 'git+https://github.com/juju/amulet'
+
+
+RUN pip2 install -U charm-tools pyopenssl
+RUN pip3 install -U pytest  pytest-asyncio asyncio_extras juju requests pyyaml kubernetes
+RUN snap install conjure-up --classic
```

Comment:
A few concerns about dependency stuff:

1. Could we clone and run `integration-tests/install-deps.sh`? That may need some updating, but if we use it in our automation regularly then we can make sure it stays relevant.
2. How will it work to upgrade dependencies on all the containers? Sometimes we need a new release of python-libjuju or whatever to fix a problem in the tests. Will that be easy?
3. juju coming from both ppa:juju/stable and the conjure-up snap might be OK, but it's a little scary. I've had weird problems with that in the past.
4. charm-tools coming from two places is also scary
5. Maybe unneeded stuff: cloud-weather-report, matrix